Exception handling refactoring

elrido opened the pull request:
In a prior PR, @Ribas160 and me [discussed](https://github.com/PrivateBin/PrivateBin/pull/1718#issuecomment-3538739167) our PHP exception handling. This is the result of that. I first only wanted to add that translating exception, so one could skip guessing if an exception message needs translation, but then also noticed that another common exception we deal with comes from JSON encoding errors. This then allows for several catches to be more explicit, which should help readability.

Finally I noticed some other areas that could be improved, such as incorrect throws-doc-blocks, adding some error logging on caught, but suppressed, exceptions, JSON encoding and translating (sic) a boolean used in the templates or restating some logic to make early, explicit return lines unnecessary.

## Changes
* Refactored translation of exception messages
* Some exceptions not getting translated
* Corrected doc-blocks
* Added some error logging on purposely suppressed exceptions that may help administrators
* Removed unnecessary JSON encoding and translation of `_is_deleted`  boolean
* more explicit catches, where possible - I left spots alone, where we handle several exception types in the same way
* prefer null coalescing operator over `array_key_exists`
* use strict equality checks
* prefer `empty` over `!== ''` for readability
* prefer `array_key_exists` over alias `key_exists` (stick to a single coding style, also made me initially miss some null coalescing operator opportunities)
* prefer `DirectoryIterator` over `dir` for readability and not having to remember to close the open handle
* check for and log file deletion errors
* test jbo translation file for completeness

If any of you can think of other opportunities or would like to experiment further, feel free to commit your work onto this branch as well.

elrido committed:
refactored exceptions in controller

- added missing exception doc blocks
- introduced exception type that translates message during construction
- catch explicit exception types where possible

elrido committed:
Refactored translation of exception messages

elrido committed:
apply StyleCI recommendation

elrido committed:
fix PHP 7.4 compatibility

elrido committed:
apply StyleCI recommendation

elrido committed:
remove unnecessary variable

Ribas160 reviewed (changes requested):
Thank you for this refactor!
I think now the code looks like more readable and standardized

elrido committed:
apply null coalescing operator

Co-authored-by: Mikhail Romanov <42250412+Ribas160@users.noreply.github.com>

elrido committed:
apply null coalescing operator

elrido committed:
apply StyleCI recommendation

elrido committed:
apply Scrutinizer fix

elrido committed:
apply null coalescing operator, strict equality, avoid aliases, prefer empty

elrido committed:
apply StyleCI recommendation

elrido committed:
prefer DirectoryIterator for readability, also test jbo translation, log deletion errors

elrido committed:
apply Scrutinizer fix

elrido commented:
For once, I'm really pleased with the results of the [Scrutinizer report](https://scrutinizer-ci.com/g/PrivateBin/PrivateBin/inspections/e5b8c68c-c1b6-4313-9cd5-3425e29ce83c) and it caught me out on two issues I had introduced. One of the earliest analyzers we introduced and it still pays dividends.
